bin/process-csv.js

```diff
@@ -31,143 +31,163 @@
+                                use_config_min_sale_qty: 1,
+                                max_cart_qty: 0,
+                                use_config_max_sale_qty: 1,
+                                is_in_stock: Math.min(1, record.NoInStock),
+                                notify_on_stock_below: '',
+                                use_config_notify_stock_qty: 1,
+                                manage_stock: 1,
+                                use_config_manage_stock: 1,
+                                qty_increments: 0,
+                                use_config_qty_increments: 1,
+                                enable_qty_increments: 0,
+                                use_config_enable_qty_inc: 1,
+                                is_decimal_divided: 0,
+                                website_id: 1,
+                                deferred_stock_update: 0,
+                                use_config_deferred_stock_update: 1,
+                                related_skus: '',
+                                crosssell_skus: '',
+                                upsell_skus: '',
+                                additional_images: '',
+                                additional_image_labels: '',
+                                hide_from_product_page: '',
+                                custom_options: ''
+                            };
+                        }))
+                        .on('error', console.error)
+                        .on('end', function () {
+                            if (skippedRecords.length > 0) {
+                                csv.stringify(
+                                    _(skippedRecords)
+                                        .map(function (skipped, index) {
+                                            return [
+                                                index + 1,
+                                                skipped.record.BookCode,
+                                                skipped.record.Title,
+                                                skipped.reason
+                                            ];
+                                        })
+                                        .unshift([ '#', 'BookCode', 'Title', 'Reason for Skipping' ])
+                                        .value(),
+                                    function (err, csvData) {
+                                        fs.writeFile(targetPath + '/skipped-records.csv', csvData);
+                                    }
+                                );
+                            }
+                        })
+                        .pipe(csv.stringify({ header: true }))
+                        .pipe(fs.createWriteStream(targetPath + '/products.csv'));
                 })
-                .pipe(csv.stringify({ header: true }))
-                .pipe(fs.createWriteStream(targetPath + '/products.csv'));
+                .pipe(csv.stringify({ header: true }));
```

Comment:
Even though it isn't being saved anywhere, this step seems to be necessary, otherwise the `end` event (above) never fires.